### PR TITLE
chore: mitigate some races in nexus child 

### PR DESCRIPTION
**Pull request by Hoverbear**
This mitigates, but does not address, the core problems described in CAS-771.

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

File: mayastor/src/grpc/mod.rs [hunk 1]
```diff
@@ -75,13 +83,15 @@ pub async fn sync_config<F, T>(future: F) -> GrpcResult<T>
 where
     F: Future<Output = GrpcResult<T>>,
 {
+    let guard = RECONFIGURING.lock().await;
```
Comment (tiagolobocastro) [inline mayastor/src/grpc/mod.rs:86]: I might be mistaken but I thought that we already only had one gRPC call at a time?

Comment (Hoverbear) [reply mayastor/src/grpc/mod.rs:86]: Maybe someone else can confirm?

Comment (tiagolobocastro) [reply mayastor/src/grpc/mod.rs:86]: Maybe @gila or @jonathan-teh know?

**Pull request closed without merging**